pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -150,6 +152,22 @@ public void testInstancesStarted() {
     }
   }
 
+  @Test
+  public void testInvalidTableConfig() {
+    TableConfig tableConfig =
+        new TableConfig.Builder(CommonConstants.Helix.TableType.OFFLINE).setTableName("badTable").build();
+    ObjectNode jsonConfig = tableConfig.toJsonConfig();
+    // Remove a mandatory field
+    jsonConfig.remove(TableConfig.VALIDATION_CONFIG_KEY);
+    try {
+      sendPostRequest(_controllerRequestURLBuilder.forTableCreate(), jsonConfig.toString());
+      fail();
+    } catch (IOException e) {
+      // Should get response code 400 (BAD_REQUEST)
+      assertTrue(e.getMessage().startsWith("Server returned HTTP response code: 400"));
```

Comment:
Can we assert on the rest of the message? (If we don't have access to the message at this level, ignore)

The reason is, currently the client does get a 400, but the message is "null" which makes it hard to debug.

Reply:
I understand your concern, but on this level we don't have access to the message. I added checks on the message in the unit test.